### PR TITLE
PE-2201 optional public client arg on get earnings

### PR DESCRIPTION
**Pull request by mihoward21**

### CODE REVIEW

File: packages/splits-sdk-react/src/hooks/data.ts [hunk 1]
```diff
@@ -86,6 +88,7 @@ export const useSplitEarnings = (
   splitAddress: string,
   includeActiveBalances?: boolean,
   erc20TokenList?: string[],
+  publicClient?: PublicClient<Transport, Chain>,
```
Comment (mihoward21) [inline packages/splits-sdk-react/src/hooks/data.ts:91]: kinda regretting this now. wish we had made an update at the major version upgrade to put all these into an options arg (so would be like `chainId, splitAddress, options`

anyway...not sure if you have any good ideas. could start that options arg now...or just leave this how i did it.

Comment (r0ohafza) [reply packages/splits-sdk-react/src/hooks/data.ts:91]: I am thinking we just throw an error if chainId does not match?
the other way we could do this is optionally have a user pass in a list of publicClients in the client constructor and we can use them to fetch the earnings

Comment (mihoward21) [reply packages/splits-sdk-react/src/hooks/data.ts:91]: i'll make that change

File: packages/splits-sdk/src/client/data.ts [hunk 2]
```diff
@@ -322,6 +322,8 @@ export class DataClient {
     distributed: FormattedTokenBalances
     activeBalances?: FormattedTokenBalances
   }> {
+    const functionPublicClient = this._publicClient
```
Comment (mihoward21) [inline packages/splits-sdk/src/client/data.ts:325]: i think fine to leave this change. actually probably safer to store it in a local variable, don't risk it getting changed halfway through the function execution.

**Review by r0ohafza — APPROVED** [review]
LGTM!

**Pull request merged**